From fc1a4d78aa80d128d7e63f10705a7d346f3d4aea Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 10 Dec 2018 10:49:49 -0600
Subject: [PATCH 16/18] Unvendor sqlite3

---
 PCbuild/python.props | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/PCbuild/python.props b/PCbuild/python.props
index d2b1991558..8fe1270225 100644
--- a/PCbuild/python.props
+++ b/PCbuild/python.props
@@ -32,7 +32,7 @@
 
     <!-- Directories of external projects. tcltk is handled in tcltk.props -->
     <ExternalsDir>$([System.IO.Path]::GetFullPath(`$(PySourcePath)externals\`))</ExternalsDir>
-    <sqlite3Dir>$(ExternalsDir)sqlite-3.14.2.0\</sqlite3Dir>
+    <sqlite3Dir>$(ExternalsDir)$(SQLITE3_DIR)\</sqlite3Dir>
     <bz2Dir>$(ExternalsDir)bzip2-1.0.6\</bz2Dir>
     <bsddbDir>$(ExternalsDir)bsddb-4.7.25.0</bsddbDir>
     <opensslDir>$(ExternalsDir)openssl-1.0.2o\</opensslDir>
-- 
2.17.2 (Apple Git-113)

