diff -urN Python-3.6.2.orig/PCbuild/build.bat Python-3.6.2/PCbuild/build.bat
--- Python-3.6.2.orig/PCbuild/build.bat	2017-08-19 20:46:25.000000000 +0100
+++ Python-3.6.2/PCbuild/build.bat	2017-08-19 20:46:40.000000000 +0100
@@ -83,7 +83,7 @@
 if "%IncludeSSL%"=="" set IncludeSSL=true
 if "%IncludeTkinter%"=="" set IncludeTkinter=true
 
-if "%IncludeExternals%"=="true" call "%dir%get_externals.bat"
+rem if "%IncludeExternals%"=="true" call "%dir%get_externals.bat"
 
 if "%do_pgo%" EQU "true" if "%platf%" EQU "x64" (
     if "%PROCESSOR_ARCHITEW6432%" NEQ "AMD64" if "%PROCESSOR_ARCHITECTURE%" NEQ "AMD64" (
